tests/hook/execute.c

```diff
@@ -0,0 +1,99 @@
+#include "clar_libgit2.h"
+
+static git_repository *g_repo = NULL;
+
+void test_hook_execute__initialize(void)
+{
+	g_repo = cl_git_sandbox_init("testrepo");
+}
+
+void test_hook_execute__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+	g_repo = NULL;
+}
+
+static int hook_exec_1(git_hook_env env, void *payload)
+{
+	int *hook_called = payload;
+
+	cl_assert_equal_i(env.args.count, 1);
+	cl_assert_equal_s(env.args.strings[0], "1");
+
+	*hook_called = 1;
+
+	return 0;
+}
+
+void test_hook_execute__hook_called(void)
+{
+	int hook_called = 0;
+	cl_must_pass(git_hook_register_callback(g_repo, hook_exec_1, NULL, &hook_called));
+	cl_must_pass(git_hook_execute(g_repo, "post-merge", "1", NULL));
+	cl_assert_equal_i_(hook_called, 1, "hook wasn't called");
+}
+
+static int hook_exec_2(git_hook_env env, void *payload)
```

Comment:
I expected this hook callback to consume two arguments. Maybe name the callbacks a bit clearer, e.g. `hook_with_argument_cb` and `hook_with_io_cb` (those names suck, I know, but I guess they're still better than enumerating the callbacks).